docs/source/library_reference.rst

```diff
@@ -182,17 +223,35 @@ The following are internal classes used within ``cocotb``.
     :member-order: bysource
     :private-members:
 
-Testbench Structure
-===================
 
-Clock
------
+Test Utilities
+==============
+
+Clock Driver
+------------
 
 .. autoclass:: cocotb.clock.Clock
     :members:
     :member-order: bysource
 
 
+Asynchronous Queues
+-------------------
+
+.. automodule:: cocotb.queue
+    :members:
+    :member-order: bysource
+    :synopsis: Asynchronous queues.
+
+
+Using Synchronous Functions
+---------------------------
+
+.. autofunction:: cocotb.external
+
+.. autofunction:: cocotb.function
+
+
```

Comment:
The plan is to keep these undocumented as we really don't want people using them. I will probably deprecate them, but leave them in cocotb until they get in the way. A replacement for `cocotb.external` that acts more like what most users want will be introduced.